### ISSUE TITLE
Why only top lidar?

### ISSUE THREAD

**Issue by fcakyon** [gdlg/simple-waymo-open-dataset-reader#1]
I have to say that, this repo is really useful for using the waymo dataset on non-tf frameworks :)
However when i look for the [visualise_labels_and_lidar](https://github.com/gdlg/simple-waymo-open-dataset-reader/blob/master/examples/visualise_labels_and_lidar.py), [groundtruth_extraction](https://github.com/gdlg/simple-waymo-open-dataset-reader/blob/master/examples/groundtruth_extraction.py) scripts I see they only use top lidar points instead of combining all lidar point clouds. Am I missing something or is it really the case?

**Comment by gdlg** [comment]
Thanks, I am glad that it’s useful.

I have never tried to combine the information of multiple LIDARs, though there is no reason it wouldn’t work. I was using the LIDAR to check which object is occluded or not.

It shouldn’t be too hard to loop through all the LIDARs and combine the information. I would be happy to accept a pull request if you have a look at it.

**Comment by fcakyon** [comment]
Thanks for the reply, I will consider doing an pr.

Another question: [At this line](https://github.com/gdlg/simple-waymo-open-dataset-reader/blob/master/examples/groundtruth_extraction.py#L70), by saying "box space" did you mean "global frame"? I'm taking the coordinate system given in [Coordinate Systems section](https://waymo.com/open/data/) as reference.

**Comment by gdlg** [comment]
No, this is a coordinate system set at the centre of a given label’s bounding box and x-axis pointing toward the front of the vehicle, y-axis toward the side and z-axis up. Each axis is scaled such that in this frame, a point is inside the bounding box if and only if all its coordinates are in the range [-1,1].

This is used [at this line](https://github.com/gdlg/simple-waymo-open-dataset-reader/blob/d83acb9fe1af8a65551b33e6b047aaf553025888/examples/groundtruth_extraction.py#L83) to determine which LIDAR point falls within the bounding box of each label and count the number of LIDAR points per label. Alas this involves a cartesian product [here](https://github.com/gdlg/simple-waymo-open-dataset-reader/blob/d83acb9fe1af8a65551b33e6b047aaf553025888/examples/groundtruth_extraction.py#L79) between each LIDAR point and each label. That’s why this code is quite slow.
